### PR TITLE
Upgrade dhall dep to 1.13

### PR DESCRIPTION
**Pull request by jneira**
To enjoy the improvements of the new version (including the useful new type diffs errors!)

### CODE REVIEW

**Comment by jneira** [comment]
Oh, yeah, thanks for remind me, i will modify the dhall file.
Do you think that dhall-to-cabal files in golden tests should be in sync?

**Comment by quasicomputational** [comment]
Sorry, I don't quite understand the question.

If you mean the cabal-to-dhall files, which I think will have been affected by how dhall 1.13 unconditionally pretty-prints, I'd suggest changing them, yes. It ought to be okay to run the tests with `--accept` and then just make sure that the results look sensible.

**Comment by ocharles** [comment]
I think @jneira is asking if when he changes the main `dhall-to-cabal.dhall` file, he should also change `golden-tests/dhall-to-cabal.dhall`. I'm not too fussed either way - a change like this doesn't exercise any different code.

File: dhall-to-cabal.cabal [hunk 1]
```diff
@@ -94,13 +94,12 @@ library
         bytestring ^>=0.10,
         containers ^>=0.5,
         contravariant ^>=1.4,
-        dhall ^>=1.12.0,
+        dhall ^>=1.13.0,
```
Comment (ocharles) [inline dhall-to-cabal.cabal]: As @quasicomputational noted, this should be done in the `dhall-to-cabal.dhall` file. I also think we should support `1.12` as well, as we don't depend on the API of 1.13.

Comment (jneira) [reply dhall-to-cabal.cabal]: Ok, with the new "majorVersions" syntax it can be expressed with `dhall ^>=1.12.0 || ^>=1.13.0`, no ?

Comment (jneira) [reply dhall-to-cabal.cabal]: Anyway, after a second thought, does not [this change in `lib/DhallToCabal.hs`](https://github.com/dhall-lang/dhall-to-cabal/pull/49/commits/c91e61804c2fddbb8007ec54ec02ab27cf6201c5#diff-d1c1b583c62f369b4b84141da6c209c2R557) make it depends on 1.13 exclusively?

**Comment by ocharles** [comment]
Looks like the CI build is broken, but I think I can fix that. We need to update the `dhall.nix` expression with an expression that would build 1.13.

**Comment by ocharles** [comment]
Ok, this has been manually merged. Thanks!

**Pull request closed without merging**